Components layout

Deman4ig opened the pull request.

Deman4ig committed:
add components library to the main page

jsru commented:
Добавляю преподавателя (@GSemikozov) для код-ревью.

GSemikozov commented on index.html:
```diff
@@ -1,25 +1,169 @@
+      <div class='components-grid__element'>
+        <div class='preview'>
+          <div class='preview__body'></div>
+          <p class='preview__header'>Custom form elements</p>
```
странно что хедер идет за боди... тогда может footer?

Deman4ig replied:
действительно странно:)

GSemikozov commented on components/preview/preview.css:
```diff
@@ -0,0 +1,23 @@
+.preview {
```
в примере на лекции это был компонент-контейнер, а поскольку у нас уже есть main__inner, этоьт компонент я бы посоветовал переименовать в .card -  так будет все-таки логичнее. Но - это не обязательно.

Deman4ig replied:
согласен - card звучит более подходящим

Deman4ig committed:
refactor component names

Deman4ig committed:
lesson-4

add logo
add nav-bar

GSemikozov commented on components/components-grid/components-grid.css:
```diff
@@ -0,0 +1,15 @@
+
+.components-grid {
+    display: flex;
+    flex-wrap: wrap;
```
здесь, по идее, еще нужно добавить отрицательные марджины margin: 0 -15px (чтобы растянуть контейнер на 15 пикселей крайних блоков слева и справа)

GSemikozov replied:
то есть правильнее это сделать здесь, а не на уровне main__inner

Deman4ig replied:
согласен!

GSemikozov commented on components/navigation/navigation.css:
```diff
@@ -0,0 +1,30 @@
+    font-weight: normal;
+    font-size: 16px;
+    line-height: 24px;
+    display: flex;
```
поскольку ссылка должна бы растягиваться по ширине внутреннего содержимого (текста), а не родителя - все-таки предпочтительнее использовать inline-flex

Deman4ig replied:
у меня эта фигня осталась в коде случайно - т.к. сначала сделал через ul > li > a, а потом подумал что наверное нет смысла в такой структуре и переделал на nav > a.

так что убрал вообще display: flex

Deman4ig committed:
Refactor components-grid and navigation links

1. Move negative margin from main__inner to components-grid
2. Remove display:flex from navigation__link element

jsru commented:
К сожалению, PR не может быть замержен, GSemikozov вероятно сможет помочь.

Deman4ig committed:
Merge branch 'master' of github.com:js-tasks-ru/htmlcss-20210513_deman4ig into components-layout

Deman4ig committed:
Merge branch 'master' of github.com:js-tasks-ru/htmlcss-20210513_deman4ig into components-layout

Deman4ig committed:
Resolve conflict in sprites.svg
